src/foam/core/ContextAware.js

```diff
@@ -17,5 +17,21 @@ foam.INTERFACE({
       type: 'Void',
       args: [ { name: 'x', type: 'Context' } ]
     }
+  ],
+
+  axioms: [
+    {
+      name: 'javaExtras',
+      buildJavaClass: function(cls) {
+        cls.methods.push(`
+          static <T> T maybeContextualize(X x, T obj) {
```

Comment:
FObject extends ContextAware, so when would obj ever not be an instanceof ContextAware?

Reply:
obj can be of any type, doesn't have to be FObject.